### PR TITLE
tony doesnt wait for chief worker to finish before stopping #134

### LINKED ISSUE #134: tony doesn't wait for chief worker to finish before stopping
We wait for all workers to finish before stopping the training job. However, `chief` is another type of job we should also wait for.

Related code snippet.
```
      if (numCompletedWorkerTasks.get() == numTotalWorkerTasks) {
        Utils.printWorkerTasksCompleted(numCompletedWorkerTasks, numTotalWorkerTasks);
        break;
      }
```

This should be configurable as well.

### PR DESCRIPTION
**Pull request by oliverhu**
Fix #134

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java [hunk 1]
```diff
@@ -178,6 +178,10 @@ public static String getContainerResourcesKey() {
   public static final String WORKER_TIMEOUT = WORKER_PREFIX + "timeout";
   public static final int DEFAULT_WORKER_TIMEOUT = 0;
 
+  // Job types that we don't wait to finish
+  public static final String IGNORED_JOBTYPES = TONY_APPLICATION_PREFIX + "ignored.jobtypes";
+  public static final String IGNORED_JOBTYPES_DEFAULT = "ps";
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/TonyConfigurationKeys.java]: Can we rename this `tony.untracked.jobtypes`?

File: tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java [hunk 2]
```diff
@@ -251,4 +251,26 @@ public void testTonyResourcesFlag() throws ParseException {
...
+   * is gonna fail.
+   */
+  @Test
+  public void testWaitingForChiefToFinish() throws ParseException {
+    conf.setBoolean(TonyConfigurationKeys.IS_SINGLE_NODE, false);
+    conf.setInt(TonyConfigurationKeys.TASK_MAX_MISSED_HEARTBEATS, 2);
+    client = new TonyClient(conf);
+    client.init(new String[]{
+        "--src_dir", "tony-core/src/test/resources/",
+        "--executes", "tony-core/src/test/resources/conditional_wait.py",
```
Comment (erwa) [inline tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java]: After https://github.com/linkedin/TonY/pull/131, we should be able to just specify `--executes conditional_wait.py`, no?

File: tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java [hunk 3]
```diff
@@ -251,4 +251,26 @@ public void testTonyResourcesFlag() throws ParseException {
     int exitCode = client.start();
     Assert.assertEquals(exitCode, 0);
   }
+
+
+  /*
+   * In conditional_wait.py script, we wait for 5s for chief job type. If we only wait for worker to finish, this test
+   * is gonna fail.
```
Comment (erwa) [inline tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java]: I don't understand why this test would fail even if we only waited for non-chief workers to finish.

File: tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java (hunk 2 above)
Comment (oliverhu) [reply tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java]: Yah, i branched off master for this fix. didn't pick up #131 . will rebase that.

File: tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java (hunk 3 above)
Comment (oliverhu) [reply tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java]: This test tries to mimic the scenario happens in #134. 

Prior to this change, we don't wait for non-chief worker to finish and conclude the training. The logic in TonySession's `updateSessionStatus()` method used to make decision based on if non-ps worker has finished, if no -> setFinalStatus false.

Comment (erwa) [reply tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java]: Got it, that makes sense. Thanks for explaining.

File: tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java [hunk 4]
```diff
@@ -251,4 +251,26 @@ public void testTonyResourcesFlag() throws ParseException {
...
+   */
+  @Test
+  public void testWaitingForChiefToFinish() throws ParseException {
+    conf.setBoolean(TonyConfigurationKeys.IS_SINGLE_NODE, false);
+    conf.setInt(TonyConfigurationKeys.TASK_MAX_MISSED_HEARTBEATS, 2);
```
Comment (erwa) [inline tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java]: Are either of these two confs necessary? IS_SINGLE_NODE=false is already the default, and the missed heartbeats setting doesn't seem relevant to this test.

Comment (oliverhu) [reply tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java]: Makes sense. Removed those 2
